tests/modules/programs/borgmatic/neither-sourcedirectories-nor-patterns.nix

```diff
@@ -0,0 +1,26 @@
+{ config, pkgs, ... }:
+
+let
+
+  backups = config.programs.borgmatic.backups;
+
+in {
+  programs.borgmatic = {
+    enable = true;
+    backups = {
+      main = {
+        location = {
+          sourceDirectories = null;
+          patterns = null;
```

Comment:
nit: From a user point of view, I think it is more likely that none of these 2 options are specified rather both being explicitly set to `null`. As a result, I would delete these 2 lines.